src/main/java/hudson/plugins/gradle/injection/InjectionUtil.java

```diff
@@ -78,4 +87,49 @@ private static boolean isEnabled(Set<String> labels, Set<String> enabledNodes) {
         }
         return enabledNodes.stream().anyMatch(labels::contains);
     }
+
+    public static boolean isVcsRepositoryAllowed(Run<?, ?> run, InjectionConfig config) {
+        if (config.getParsedInjectionVcsRepositoryPatterns() == null || config.getParsedInjectionVcsRepositoryPatterns().isEmpty()) {
+            return true;
+        }
+
+        if (run instanceof WorkflowRun) {
+            Collection<? extends SCM> scms = ((WorkflowRun) run).getParent().getSCMs();
+
+            for (SCM scm : scms) {
+                if (vcsRepositoryUrlMatches(config, scm)) {
+                    return true;
+                }
+            }
+        } else if (run instanceof AbstractBuild) {
+            AbstractProject<?, ?> project = ((AbstractBuild<?, ?>) run).getProject();
+            SCM scm = project.getScm();
+
+            if (scm == null) {
+                return true;
+            }
+
+            return vcsRepositoryUrlMatches(config, scm);
+        }
+
+        return false;
```

Comment:
Is it worth covering this with an additional `try { ... } catch {...}` clause to avoid any potential errors? Or perhaps do this in our EnvironmentContributors (for Gradle and Maven)?

The build will fail if an error happens here for whatever reason.